src/FileManager.cpp

```diff
@@ -37,7 +37,61 @@ void FileManager::setCurrentFileName(const QString fileName)
 
 void FileManager::newFile()
 {
-    // Logic to create a new file
+    QString currentFileName = getCurrentFileName();
+    bool isFileSaved        = !currentFileName.isEmpty();
+    bool isTextEditorEmpty  = this->m_editor->toPlainText().isEmpty();
+    // File has not been saved and the text editor is not empty
+    if (!isFileSaved && !isTextEditorEmpty)
+    {
+        // Create box to prompt user to save changes to file
+        QMessageBox promptBox;
+        promptBox.setWindowTitle("Save Current File");
+        promptBox.setText("Would you like to save the file?");
+        promptBox.setStandardButtons(QMessageBox::Save | QMessageBox::Cancel);
+        promptBox.setDefaultButton(QMessageBox::Save);
+
+        int option = promptBox.exec();
+        // return if the user hit Cancel button
+        if (option == QMessageBox::Cancel)
+        {
+            return;
+        }
+
+        saveFile();
+    }
+    // File has been previously saved
+    else if (isFileSaved)
+    {
+        // Read from saved file and compare to current file
+        QFile file(currentFileName);
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+            return;
+        QTextStream in(&file);
+        QString savedFileContents = in.readAll();
+        file.close();
+        if (savedFileContents != this->m_editor->toPlainText().trimmed())
+        {
+            // Create box to prompt user to save changes to file
+            QMessageBox promptBox;
+            promptBox.setWindowTitle("Changes Detected");
+            promptBox.setText("Would you like to save the current changes to the file?");
+            promptBox.setStandardButtons(QMessageBox::Save | QMessageBox::Cancel);
+            promptBox.setDefaultButton(QMessageBox::Save);
+            int option = promptBox.exec();
+            // return if the user hit Cancel button
+            if (option == QMessageBox::Cancel)
+            {
+                return;
+            }
+            saveFile();
+        }
+    }
+    else {
+        setCurrentFileName("");
+        m_editor->clear();
+        m_mainWindow->setWindowTitle("Code Astra ~ untitled");
+    }
```

Comment:
As mentioned in the comment in the conversation, this would be better in that case (please, see the latest comment I left after you updated from my previous review):
```cpp
    if( !m_currentFileName.isEmpty())
    {
        setCurrentFileName("");
        m_editor->clear();
        m_mainWindow->setWindowTitle("Code Astra ~ untitled");
    }
```